src/MiniJuvix/Pipeline/EntryPoint.hs

```diff
@@ -1,15 +1,22 @@
-module MiniJuvix.Pipeline.EntryPoint where
+module MiniJuvix.Pipeline.EntryPoint
+  ( module MiniJuvix.Pipeline.EntryPoint,
+  )
+where
 
 import MiniJuvix.Prelude
 
--- | The head of _entryModulePaths is assumed to be the Main module.
+-- | The head of _entryModulePaths is assumed to be the Main module
 data EntryPoint = EntryPoint
-  { _entryRoot :: FilePath,
-    _entryModulePaths :: NonEmpty FilePath
+  { _entryPointRoot :: FilePath,
+    _entryPointNoTermination :: Bool,
+    _entryPointModulePaths :: NonEmpty FilePath
   }
   deriving stock (Eq, Show)
 
+defaultEntryPoint :: FilePath -> EntryPoint
+defaultEntryPoint mainFile = EntryPoint "." False (pure mainFile)
```

Comment:
now that we are adding more fields, I think it makes sense to use record syntax with explicit fields